### PR TITLE
[AAP-8159] Add Execution Environment > Templates List

### PR DESCRIPTION
**Pull request by daphnemaeve**
Adds Template List to Execution Environment page.

Before:
Screenshot 2024-02-05 at 4 41 01 PM

After:
Screenshot 2024-02-05 at 4 39 47 PM

### CODE REVIEW

**Review by prat98 — COMMENTED** [review]
I noticed in the description of https://issues.redhat.com/browse/AAP-8159, there is a search key listed for `or_type`. I also see the option to search by `type` in the old UI. Was wondering if we could add that to this page as well

File: frontend/awx/resources/templates/TemplatesList.tsx [hunk 1]
```diff
@@ -27,13 +27,18 @@ export function TemplatesList(props: {
...
+  executionEnvironmentId?: string | undefined;
 }) {
   const { t } = useTranslation();
   const pageNavigate = usePageNavigate();
   const getPageUrl = useGetPageUrl();
   const toolbarFilters = useTemplateFilters();
   const tableColumns = useTemplateColumns();
-  const getQueryParams = (projectId?: string | undefined, inventoryId?: string | undefined) => {
+  const getQueryParams = (
+    projectId?: string | undefined,
```
Comment (kialam) [inline frontend/awx/resources/templates/TemplatesList.tsx:38]: `undefined` shouldn't be necessary if the prop is optional.

**Review by kialam — APPROVED** [review]
Once the unused prop type is removed this should be good to go. Pulled down and ran locally.

**Review by kialam — APPROVED** [review]
🚀

**Pull request merged**